dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/Audit.java

```diff
@@ -80,6 +82,49 @@
     @JsonProperty
     private Object data;
 
+    @JsonIgnore
+    private AuditableEntity auditableEntity;
+
+    /**
+     * This constructor is used to create an instance of Audit using the 'Builder'
+     * pattern This should be used by a service that needs to create a new Audit
+     * entry
+     */
+    @Builder( builderClassName = "AuditBuilder" )
```

Comment:
Yes, so this is the general idea:
If you look at the Audit class, now it contains a new property:

```
@JsonIgnore
private AuditableEntity auditableEntity;
```
This new property is used to "delay" the serialization of the entity to the `AuditManager`.
This property is meant to be used only by a Builder that is actually constructing the object (for instance all the Hibernate listeners). 
The other property:

```
@JsonProperty
 private Object data;
```
is meant to be used in a constructor only by Jackson.
So, my idea was to "hide" the `.data(...)` property of the Builder (because, effectively it should not be used) by adding the first constructor. 
The second constructor (the one annotated with `@JsonCreator` and using the `data` attribute) is only supposed to be used by Jackson.

I'll remove both constructors, but we just need to be sure that the `.data(..)` property is not used (unless you know another way, using Jackson +  Lombok)

Reply:
Ok, I see. Yes, I think the only way to "hide" a field currently in a lombok builder is to manually add the builder with the fields we want (sadly this means we have to add new properties in two places).

Could we make the setter private? with something like `@Setter(AccessLevel.PRIVATE) private Object data` or maybe we still need it to be public in some places?